hazelcast/src/main/java/com/hazelcast/internal/partition/impl/MigrationStats.java

```diff
@@ -58,9 +64,11 @@
     /**
      * Marks start of new repartitioning.
      * Resets stats from previous repartitioning round.
+     * @param migrations number of planned migration tasks
      */
-    void markNewRepartition() {
+    void markNewRepartition(int migrations) {
```

Comment:
Is it guaranteed, that getters will be called from the same thread as this method, or strictly after that method is completed? I am thinking about the following sequence of events:
T1: `markNewReparatition: plannedMigrations = migrations`
T2: read new `plannedMigrations`
T2; read old `completedMigrations`
T1: `markNewReparatition: completedMigrations.set(0)`

So if thread-safety or strict consistency is a concern here, we'd better employ another volatile holder of the state here.

Reply:
- For the migration events, yes, reading thread is the same as modifying thread.
- For migration logging, readers will read after this method is returned (after migration tasks are scheduled). During migration, stats are updated separately, such as `completedMigrations` or `elapsedMigrationTime`. It will be eventually consistent with monotonicity.  
- For metrics system, what you described is possible. It reads the stats periodically without any order guarantees. But not sure if we need to more complexity for metrics...

What do you think @mmedenjak ?

Reply:
An additional complication for metrics is typically not needed. Given that the consistency of events is not compromised, please let me revoke that comment.